validmind/tests/model_validation/statsmodels/ScorecardHistogram.py

```diff
@@ -53,120 +49,88 @@ class ScorecardHistogram(Metric):
-
-        fig = make_subplots(rows=1, cols=2, subplot_titles=("Train Data", "Test Data"))
-
-        trace_train_0 = go.Histogram(
-            x=scores_train_0, opacity=0.75, name=f"Train {target_col} = 0"
-        )
-        trace_train_1 = go.Histogram(
-            x=scores_train_1, opacity=0.75, name=f"Train {target_col} = 1"
+    def plot_score_histogram(dataframes, dataset_titles, score_col, target_col, title):
+        figures = []
+        # Define a color map for multiple classes
+        class_colors = {
+            0: "blue",
+            1: "red",
+            2: "green",
+            3: "purple",
+            4: "orange",
+            # ... add more colors for more classes
+        }
+
+        # Check if there are more unique class values than colors defined
+        unique_classes = set(
+            np.concatenate([df[target_col].unique() for df in dataframes])
         )
-        trace_test_0 = go.Histogram(
-            x=scores_test_0, opacity=0.75, name=f"Test {target_col} = 0"
-        )
-        trace_test_1 = go.Histogram(
-            x=scores_test_1, opacity=0.75, name=f"Test {target_col} = 1"
-        )
-
-        fig.add_trace(trace_train_0, row=1, col=1)
-        fig.add_trace(trace_train_1, row=1, col=1)
-        fig.add_trace(trace_test_0, row=1, col=2)
-        fig.add_trace(trace_test_1, row=1, col=2)
-
-        fig.update_layout(barmode="overlay", title_text=title)
-
-        return fig
+        if not unique_classes.issubset(class_colors.keys()):
+            missing_classes = unique_classes - set(class_colors.keys())
+            raise ValueError(
+                f"Color not defined for classes: {missing_classes}. Please update the class_colors dictionary."
+            )
+
+        for i, (df, dataset_title) in enumerate(zip(dataframes, dataset_titles)):
+            fig = go.Figure()
+            for class_value in sorted(df[target_col].unique()):
+                scores_class = df[df[target_col] == class_value][score_col]
+                fig.add_trace(
+                    go.Histogram(
+                        x=scores_class,
+                        opacity=0.75,
+                        name=f"{dataset_title} {target_col} = {class_value}",
+                        marker=dict(color=class_colors[class_value]),
+                    )
+                )
+            fig.update_layout(
+                barmode="overlay",
+                title_text=f"{title} - {dataset_title}",
+                xaxis_title="Score",
+                yaxis_title="Frequency",
+                legend_title=target_col,
+            )
+            figures.append(fig)
+        return figures
 
     def run(self):
-        model = (
-            self.inputs.model[0]
-            if isinstance(self.inputs.model, list)
-            else self.inputs.model
+        title = self.params.get("title", self.default_params["title"])
+        score_column = self.params.get(
```

Comment:
This should already give you the value of `score_column` since a default value is defined:


```py
score_column = self.params["score_column"]
```

As a next step I suggest checking if the score column exists in the dataset and throw an error if it doesn't exist. It will catch users by surprise when they try to run the test without passing any parameters and see that we automatically try to access a `score` column.